test/index.spec.ts

```diff
@@ -1,8 +1,22 @@
 // Import Third-party Dependencies
 import { expect } from "chai";
+import { buildURL } from "../src/scorecard";
+import { CONSTANTS } from "../src/utils";
 
-describe("do the work here!", () => {
-  it("should work", () => {
-    expect(1).to.equal(1);
+// CONSTANTS
+const defaultrepository = "NodeSecure/scanner";
+
+describe("buildURL", () => {
+  it("should construct a valid url for the result method", () => {
+    const url = buildURL(defaultrepository).href;
+
+    const expectedUrl = `https://api.securityscorecards.dev/projects/${CONSTANTS.defaultPlatform}/${defaultrepository}`;
+    expect(url).to.equal(expectedUrl);
+  });
+
+  it("should construct a valid url for the badge method", () => {
+    const url = buildURL(`${defaultrepository}/badge`).href;
+    const expectedUrl = `https://api.securityscorecards.dev/projects/${CONSTANTS.defaultPlatform}/${defaultrepository}/badge`;
+    expect(url).to.equal(expectedUrl);
   });
 });
```

Comment:
You need to add undici test (with the Mock API).

- https://fusebit.io/blog/undici-mocking/?utm_source=www.google.com&utm_medium=referral&utm_campaign=none

You can also check NodeSecure/vulnera tests (we have test here)